execution/contexts/identify_context.go

```diff
@@ -0,0 +1,61 @@
+package contexts
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/burrow/acm/validator"
+
+	"github.com/hyperledger/burrow/acm/acmstate"
+	"github.com/hyperledger/burrow/execution/errors"
+	"github.com/hyperledger/burrow/execution/exec"
+	"github.com/hyperledger/burrow/execution/registry"
+	"github.com/hyperledger/burrow/logging"
+	"github.com/hyperledger/burrow/txs/payload"
+)
+
+type IdentifyContext struct {
+	NodeWriter   registry.ReaderWriter
+	StateReader  acmstate.Reader
+	ValidatorSet validator.Reader
+	Logger       *logging.Logger
+	tx           *payload.IdentifyTx
+}
+
+func (ctx *IdentifyContext) Execute(txe *exec.TxExecution, p payload.Payload) error {
+	var ok bool
+	ctx.tx, ok = p.(*payload.IdentifyTx)
+	if !ok {
+		return fmt.Errorf("payload must be IdentifyTx, but is: %v", txe.Envelope.Tx.Payload)
+	}
+
+	// validators are tightly bound to input accounts
+	account, err := ctx.StateReader.GetAccount(ctx.tx.Input.Address)
+	if err != nil {
+		return err
+	} else if account == nil {
+		ctx.Logger.InfoMsg("cannot find input account",
+			"tx_input", ctx.tx.Input)
+		return errors.ErrorCodeInvalidAddress
+	}
+
+	// we allow identification of non-bonded accounts
+	// as long as they have the correct permission
+	if !hasBondPermission(ctx.StateReader, account, ctx.Logger) {
+		ctx.Logger.InfoMsg("account lacks bond permission",
+			"address", account.Address)
+		return errors.ErrorCodeNoBondPermission
+	}
+
+	data, err := ctx.tx.Node.Marshal()
+	if err != nil {
+		return fmt.Errorf("could not marshal node: %v", err)
+	}
+
+	// if we are here then TxInput is correctly signed by the accounts public key
+	valid := ctx.tx.Node.PublicKey.TendermintPubKey().VerifyBytes(data, ctx.tx.Signature)
```

Comment:
Hm, so we require that:

a) A validator, or could-be-a-validator must sign the tx, and
b) The node must sign the node payload of the tx with its node key.

And now I see:

c) the input is expected to be the validator address of the node registering, kind of.